src/sql/parts/notebook/outputs/renderers.ts

```diff
@@ -265,6 +271,88 @@ export function renderLatex(
 	return Promise.resolve(undefined);
 }
 
+/**
+ * Render DataResource as a grid into a host node.
+ *
+ * @params options - The options for rendering.
+ *
+ * @returns A promise which resolves when rendering is complete.
+ */
+export function renderDataResource(
+	options: renderDataResource.IRenderOptions
+): Promise<void> {
+	// Unpack the options.
+	let { host, source } = options;
+	let sourceObject: IDataResource = JSON.parse(source);
+
+	// Before doing anything, avoid re-rendering the table multiple
+	// times (as can be the case when going untrusted -> trusted)
+	while (host.firstChild) {
```

Comment:
If you've scrolled down the grid and change trusted -> untrusted is this going to reset so we go to the top of the results? Should we have some way of avoiding re-rendering unnecessarily?

Reply:
Not from what I could see; we'd need to differentiate these cases, and that's nontrivial :(.